docs: Add Author's GitHub Profile Link to README (#5)
omerfdmrl/omerfdmrl: README.md

# [Ailemdar](https://github.com/ailemdar)

> [![GitHub Actions](https://img.shields.io/badge/GitHub%20Actions-Ready-blue?logo=github-actions)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

This badge link is incorrect. The 'GitHub Actions' badge should link to the project's GitHub Actions workflow page or the repository itself, not a personal GitHub profile. All subsequent badge links appear to have the same issue.

**Suggestion:**
```suggestion
Change `](https://github.com/omerfdmrl)` to `](https://github.com/ailemdar/actions)` or `](https://github.com/ailemdar)` as appropriate for the badge's context.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![GitHub Actions](https://img.shields.io/badge/GitHub%20Actions-Ready-blue?logo=github-actions)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

The 'GitHub Actions' badge link `https://github.com/omerfdmrl` is incorrect. This should link to the project's GitHub Actions workflow page (e.g., `https://github.com/ailemdar/ailemdar/actions`) or the specific workflow file, not a personal profile. All badges should link to relevant project resources.

**Suggestion:**
```suggestion
Change `[![GitHub Actions](https://img.shields.io/badge/GitHub%20Actions-Ready-blue?logo=github-actions)](https://github.com/omerfdmrl)` to `[![GitHub Actions](https://img.shields.io/badge/GitHub%20Actions-Ready-blue?logo=github-actions)](https://github.com/ailemdar/ailemdar/actions)` (assuming 'ailemdar/ailemdar' is the project repo).
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![Python](https://img.shields.io/badge/Python-3.11+-blue?logo=python)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

This badge link is incorrect. The 'Python' badge should link to information about the Python version used in the project, or the project's repository, not a personal GitHub profile.

**Suggestion:**
```suggestion
Change `](https://github.com/omerfdmrl)` to `](https://github.com/ailemdar)` or remove the link if it's purely informational.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![Python](https://img.shields.io/badge/Python-3.11+-blue?logo=python)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

The 'Python' badge link `https://github.com/omerfdmrl` is incorrect. This should ideally link to Python's official website, a specific project requirement document, or be removed if not providing useful context. Linking to a personal profile is misleading.

**Suggestion:**
```suggestion
Consider changing `[![Python](https://img.shields.io/badge/Python-3.11+-blue?logo=python)](https://github.com/omerfdmrl)` to `[![Python](https://img.shields.io/badge/Python-3.11+-blue?logo=python)](https://www.python.org/)` or `[![Python](https://img.shields.io/badge/Python-3.11+-blue?logo=python)](./docs/requirements.md)` if you have a specific requirements document.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![License](https://img.shields.io/badge/License-MIT-green.svg)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

This badge link is incorrect. The 'License' badge should link to the project's license file (e.g., `LICENSE`), not a personal GitHub profile.

**Suggestion:**
```suggestion
Change `](https://github.com/omerfdmrl)` to `](https://github.com/ailemdar/blob/main/LICENSE)`.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![License](https://img.shields.io/badge/License-MIT-green.svg)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

The 'License' badge link `https://github.com/omerfdmrl` is incorrect. This should link directly to the project's `LICENSE` file within the repository (e.g., `https://github.com/ailemdar/ailemdar/blob/main/LICENSE`).

**Suggestion:**
```suggestion
Change `[![License](https://img.shields.io/badge/License-MIT-green.svg)](https://github.com/omerfdmrl)` to `[![License](https://img.shields.io/badge/License-MIT-green.svg)](./LICENSE)` or `[![License](https://img.shields.io/badge/License-MIT-green.svg)](https://github.com/ailemdar/ailemdar/blob/main/LICENSE)`.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![Docker](https://img.shields.io/badge/Docker-Ready-blue?logo=docker)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

This badge link is incorrect. The 'Docker' badge should link to the project's Docker setup instructions, Docker Hub page, or the repository, not a personal GitHub profile.

**Suggestion:**
```suggestion
Change `](https://github.com/omerfdmrl)` to `](https://github.com/ailemdar)` or a relevant Docker documentation link.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![Docker](https://img.shields.io/badge/Docker-Ready-blue?logo=docker)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

The 'Docker' badge link `https://github.com/omerfdmrl` is incorrect. This should link to the project's Docker Hub page, a Dockerfile in the repository, or documentation on how to use Docker with the project.

**Suggestion:**
```suggestion
Change `[![Docker](https://img.shields.io/badge/Docker-Ready-blue?logo=docker)](https://github.com/omerfdmrl)` to `[![Docker](https://img.shields.io/badge/Docker-Ready-blue?logo=docker)](https://hub.docker.com/r/your-org/ailemdar)` or `[![Docker](https://img.shields.io/badge/Docker-Ready-blue?logo=docker)](./Dockerfile)`.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![BitBucket](https://img.shields.io/badge/BitBucket-Supported-blue?logo=bitbucket)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

This badge link is incorrect. The 'BitBucket' badge should link to relevant BitBucket integration documentation or the project's repository, not a personal GitHub profile.

**Suggestion:**
```suggestion
Change `](https://github.com/omerfdmrl)` to `](https://github.com/ailemdar)` or a relevant BitBucket documentation link.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> [![BitBucket](https://img.shields.io/badge/BitBucket-Supported-blue?logo=bitbucket)](https://github.com/omerfdmrl)

Author: **BUG-RISK** (critical)

The 'BitBucket' badge link `https://github.com/omerfdmrl` is incorrect. This should link to relevant BitBucket documentation, the project's BitBucket repository (if applicable), or a page detailing BitBucket integration.

**Suggestion:**
```suggestion
Change `[![BitBucket](https://img.shields.io/badge/BitBucket-Supported-blue?logo=bitbucket)](https://github.com/omerfdmrl)` to a more appropriate link, or remove the badge if BitBucket support is not a primary feature to highlight with a direct link.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

> 
> Automated PR reviews and issue resolution powered by state-of-the-art LLMs

**Author:** [omerfdmrl](https://github.com/omerfdmrl)

---

## Features

- **AI-Powered PR Reviews** - Get intelligent code reviews with inline comments, identifying bugs, security issues, and improvements
- **Automatic Issue Resolution** - Analyzes GitHub issues and automatically creates fixes with pull requests
- **Multi-Platform Support** - Works with both GitHub and BitBucket (Cloud and Server)
- **Multi-LLM Support** - Works with OpenAI, Anthropic Claude, Google Gemini, Ollama (local), and custom endpoints
- **Smart Context Management** - Vector-based code search using FAISS for understanding large codebases
- **Fast & Efficient** - Local embeddings support for reduced API costs and faster processing
- **Security First** - Built-in security checks and safe code modification practices
- **Structured Logging** - Comprehensive observability with structured JSON logging
- **Docker Ready** - Containerized for easy deployment as GitHub Action or standalone service

---

## Quick Start

### Using as GitHub Action

```yaml
name: AI Code Review

on:
  pull_request:

>     types: [opened, synchronize]

Author: **STYLE** (warning)

The `uses: your-org/ailemdar@v1` line in the GitHub Action example uses a generic placeholder. For a project's own README, it's better to use the actual repository path (e.g., `uses: ailemdar/ailemdar@v1`) or explicitly state that `your-org` is a placeholder for the user's organization.

**Suggestion:**
```suggestion
Consider changing `uses: your-org/ailemdar@v1` to `uses: ailemdar/ailemdar@v1` (if that's the actual repo) or add a comment like `# Replace 'your-org' with your GitHub organization or username`.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

  issues:
    types: [opened, labeled]

jobs:
  review:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      
      - name: AI Code Review
        uses: your-org/ailemdar@v1
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          llm_provider: openai
          llm_model: gpt-4o
          llm_api_key: ${{ secrets.OPENAI_API_KEY }}
```

### Local Development

> 

Author: **BUG-RISK** (warning)

The GitHub Action example uses a placeholder `your-org/ailemdar@v1`. This should be updated to the actual repository path for the action to be usable.

**Suggestion:**
```suggestion
uses: <actual-org>/ailemdar@v1
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*

```bash
# Clone the repository
git clone https://github.com/your-org/ailemdar.git
cd ailemdar

# Setup virtual environment and install dependencies
make setup-venv
source venv/bin/activate

# Configure environment
cp .env.example .env
# Edit .env with your API keys

# Run tests
make test

# Start development server
make dev
```

---

## Requirements

- **Python**: 3.11 or higher
- **Repository Access**: GitHub Token OR BitBucket credentials
- **LLM API Key**: OpenAI, Anthropic, Google, or custom provider
- **Docker**: (optional) For containerized deployment

---

## Configuration

### Platform Selection

Ailemdar supports both **GitHub** and **BitBucket**. The platform is auto-detected based on your environment variables:

- If BitBucket variables are set, BitBucket is used

> - Otherwise, GitHub is used (

Author: **BUG-RISK** (critical)

The 'Configuration' section ends abruptly with an incomplete sentence. This indicates a copy-paste error or an unfinished thought, making the documentation unclear and unprofessional.

**Suggestion:**
```suggestion
Complete the sentence, e.g., 'Otherwise, GitHub is used (default).' or provide the full context of the parenthetical statement.
```

---
*Posted by ailemdar <omerfdmrl@gmail.com>*
